File: resources/[np-core]/np-db/server/sv_db.js
// np-db/server/sv_db.js - Database bridge for NoPixel 4.0
// Wrapped in IIFE to avoid 'resourceName' clash when loaded as @np-db/server/lib.js
;(function() {
  const _resName = GetCurrentResourceName();
  setImmediate(() => console.log(`[${_resName}] DB bridge ready`));
  exports('Execute',     (q, p, cb) => exports.oxmysql.execute(q, p||[], cb||function(){}));
  exports('FetchAll',    (q, p, cb) => exports.oxmysql.execute(q, p||[], r => cb && cb(r||[])));
  exports('FetchScalar', (q, p, cb) => exports.oxmysql.scalar(q,  p||[], cb||function(){}));
  exports('Insert',      (q, p, cb) => exports.oxmysql.insert(q,  p||[], cb||function(){}));
  exports('ExecuteAsync', (q, p) => new Promise(resolve => exports.oxmysql.execute(q, p||[], r => resolve(r||[]))));
  exports('ScalarAsync',  (q, p) => new Promise(resolve => exports.oxmysql.scalar(q, p||[], r => resolve(r))));
  exports('InsertAsync',  (q, p) => new Promise(resolve => exports.oxmysql.insert(q, p||[], r => resolve(r))));
})();

File: GTAServerFXS/server-data/resources/[np-core]/np-base/database/sv_db.lua
NPX.DB = NPX.DB or {}
function NPX.DB.Execute(self, query, params, cb)
    exports["oxmysql"]:execute(query, params or {}, cb or function() end)
end
function NPX.DB.FetchAll(self, query, params, cb)
    exports["oxmysql"]:execute(query, params or {}, function(r) if cb then cb(r or {}) end end)
end
function NPX.DB.FetchScalar(self, query, params, cb)
    exports["oxmysql"]:scalar(query, params or {}, function(r) if cb then cb(r) end end)
end
function NPX.DB.Insert(self, query, params, cb)
    exports["oxmysql"]:insert(query, params or {}, function(id) if cb then cb(id) end end)
end
function NPX.DB.FetchSync(self, query, params)
    local p = promise.new()
    exports["oxmysql"]:execute(query, params or {}, function(r) p:resolve(r) end)
    return Citizen.Await(p)
end
function NPX.DB.InsertSync(self, query, params)
    local p = promise.new()
    exports["oxmysql"]:insert(query, params or {}, function(r) p:resolve(r) end)
    return Citizen.Await(p)
end
exports("Execute",     function(q,p,cb) NPX.DB:Execute(q,p,cb) end)
exports("FetchAll",    function(q,p,cb) NPX.DB:FetchAll(q,p,cb) end)
exports("FetchScalar", function(q,p,cb) NPX.DB:FetchScalar(q,p,cb) end)
exports("Insert",      function(q,p,cb) NPX.DB:Insert(q,p,cb) end)
exports("FetchSync",   function(q,p) return NPX.DB:FetchSync(q,p) end)
exports("InsertSync",  function(q,p) return NPX.DB:InsertSync(q,p) end)